### PR TITLE
Introduce is_tumbleweed function and use it for needles

### PR DESCRIPTION
**Pull request by coolo**
The tumbleweed stagings are just called Staging: - the leap ones
start with 42:

### CODE REVIEW

**Comment by coolo** [comment]
Smoke test: https://tortuga.suse.de/tests/553

File: lib/main_common.pm [hunk 1]
```diff
@@ -88,6 +89,11 @@ sub logcurrentenv {
     }
 }
 
+sub is_tumbleweed {
+    return 1 if check_var('VERSION', 'Tumbleweed');
+    return check_var('DISTRI', 'opensuse') && get_var('VERSION') =~ /^Staging:[A-Z]/;
```
Comment (okurz) [inline lib/main_common.pm]: Can you put a comment here why the staging needs to be handled as part of is_Tumbleweed?

Comment (coolo) [inline lib/main_common.pm]: the comment I would write is trivial - so what do you have in mind?

Comment (okurz) [inline lib/main_common.pm]: Let me try:

```
# staging tests are always considered Tumbleweed regardless of the target distribution
```

does this make sense?

Comment (okurz) [inline lib/main_common.pm]: Good, then move the commit comment into the file. The commit message should describe the need for change, the in file comments can describe what the code should do, isn't it?

File: products/opensuse/main.pm [hunk 2]
```diff
@@ -143,6 +141,20 @@ sub is_server() {
     return (get_var("OFW") || check_var("FLAVOR", "Server-DVD"));
 }
 
+sub is_tumbleweed {
+    # Tumbleweed and its stagings
+    return 0 unless check_var('DISTRI', 'opensuse');
```
Comment (coolo) [inline products/opensuse/main.pm]: I guess this pointless in products/opensuse - but I felt better with having this extra backup :)

File: products/opensuse/main.pm [hunk 3]
```diff
@@ -67,7 +67,7 @@ sub cleanup_needles() {
     if (!get_var('LEAP')) {
```
Comment (DimStar77) [inline products/opensuse/main.pm]: Not touched by your patch - but for consistent style it might be worthy to use if (!is_leap) here

Comment (coolo) [inline products/opensuse/main.pm]: that's what we set the LEAP variable for - to avoid having to check 4 others :)

Comment (coolo) [inline products/opensuse/main.pm]: (the unregister hook is called after main.pm ran)

**Comment by DimStar77** [comment]
LGTM - @okurz might still be missing the comment in the .pm files about why the specific "Staging:" matches TW

**Pull request merged**